python/packages/kagent-adk/src/kagent/adk/_agent_executor.py

```diff
@@ -118,9 +118,12 @@ async def execute(
                     final=False,
                 )
             )
-
+        # propagate token to session state for MCP tools. make it temporary so it won't be serialized.
+        token = context.call_context.state.get("token", None)
         # Handle the request and publish updates to the event queue
         runner = await self._resolve_runner()
+        if self._config.propagate_token and token:
+            runner.session_service = wrap_session(runner.session_service, token)
```

Comment:
Is there any harm to just always using this wrapped service? Might make the code a bit cleaner, also then we could just use normal inheritance